presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -229,11 +226,20 @@ else if (tableName.equals(Table.TIME_DIM.getName())) {
             tpcdsTable = Optional.of(Table.TIME_DIM);
         }
 
-        if (!partitioningColumn.isPresent()) {
-            throw new IllegalArgumentException(format("Table %s does not exist", tableHandle.getTableName()));
+        ConnectorTableLayout layout;
+        if (partitioningColumn.isPresent()) {
+            layout = createLayoutForPartitioningColumn(tableHandle, partitioningColumn.get(), tpcdsTable.get());
+        }
+        else {
+            layout = new ConnectorTableLayout(
```

Comment:
Is this for the Inventory layout? Maybe add a comment explaining since the next ready won't have the context that Inventory was removed from the massive if else block.